prow/pod-utils/wrapper/options.go

```diff
@@ -78,49 +83,77 @@ func (o *Options) Validate() error {
 	return nil
 }
 
-func WaitForMarker(ctx context.Context, path string) (int, error) {
+func WaitForMarkers(ctx context.Context, paths ...string) map[string]MarkerResult {
 	// Only start watching file events if the file doesn't exist
 	// If the file exists, it means the main process already completed.
-	if _, err := os.Stat(path); os.IsNotExist(err) {
-		watcher, err := fsnotify.NewWatcher()
-		if err != nil {
-			return -1, fmt.Errorf("new fsnotify watch: %v", err)
-		}
-		defer watcher.Close()
-		dir := filepath.Dir(path)
-		if err := watcher.Add(dir); err != nil {
-			return -1, fmt.Errorf("add %s to fsnotify watch: %v", dir, err)
+
+	results := make(map[string]MarkerResult)
+
+	if len(paths) == 0 {
+		return results
+	}
+
+	for _, path := range paths {
+		if _, err := os.Stat(path); !os.IsNotExist(err) {
+			results[path] = readMarkerFile(path)
 		}
+	}
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		populateMapWithError(results, fmt.Errorf("new fsnotify watch: %v", err), paths...)
+		return results
+	}
+	defer watcher.Close()
+	dir := filepath.Dir(paths[0])
```

Comment:
Yeah this is certainly assumption here.  It should be the case that all marker files and process logs are in the same directory, but yes it would be good to address this here too.